sharktank/sharktank/layers/linear.py

```diff
@@ -74,7 +83,12 @@ def forward(self, x):
         elif qdq_input is not None:
             x = qdq_input.quantize(x).unpack().dequant()
 
-        y = ops.linear(x, weight, bias)
+        if self.quarot_transform:
+            x = self.quarot_transform.forward(x)
+            x_quantized = self.quarot_transform.quantizer.quantize(x)
```

Comment:
We do not need a second input quantization. Hadamard rotation should be applied before call to `q_input` or `qdq_input`